lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -244,17 +244,55 @@ def test_duplicate(
         monkeypatch.setattr(Tarball, "_get_metadata", fake_get_metadata)
         cm = CacheManager(server_config, make_logger)
 
-        # Create a tarball file in the expected destination directory
+        # Create a tarball file in the expected destination directory: the
+        # subsequent create should report a duplicate.
         controller = cm.archive_root / "ABC"
         controller.mkdir()
         (controller / source_tarball.name).write_text("Send in the clones")
-
-        # Attempting to create a dataset from the md5 file should result in
-        # a duplicate dataset error
         with pytest.raises(DuplicateTarball) as exc:
             cm.create(source_tarball)
         assert exc.value.tarball == Dataset.stem(source_tarball)
 
+    @pytest.mark.parametrize("allow", (".md5", ""))
+    def test_move_fails(
+        self, monkeypatch, selinux_disabled, server_config, make_logger, tarball, allow
+    ):
+        src: list[Path] = []
+        dest: list[Path] = []
+        real_move = shutil.move
+
+        def mymove(source: Path, destination: Path, *args, **kwargs) -> Path:
+            src.append(source)
+            if destination.is_dir():
+                d = destination / source.name
+            else:
+                d = destination
+            dest.append(d)
+            if source.suffix == allow:
+                return real_move(source, destination, *args, **kwargs)
+            raise Exception("I'm broken")
+
+        ulink: list[Path] = []
+        ok: list[bool] = []
+
+        def unlink(self, missing_ok: bool = False):
+            ulink.append(self)
+            ok.append(missing_ok)
+
+        source_tarball, source_md5, md5 = tarball
+        monkeypatch.setattr(Tarball, "_get_metadata", fake_get_metadata)
+        cm = CacheManager(server_config, make_logger)
+        real_move = shutil.move
+        monkeypatch.setattr("pbench.server.cache_manager.shutil.move", mymove)
+        monkeypatch.setattr(Path, "unlink", unlink)
+        with pytest.raises(Exception) as e:
+            cm.create(source_tarball)
+        assert str(e.value) == "I'm broken"
+        assert src == [source_md5] + ([source_tarball] if allow else [])
+        for i in range(len(ulink)):
+            assert ulink[i] == dest[i]
+        assert ok
```

Comment:
Is the assertion at line 294 really doing what you want?  It's just checking that `ok` is not empty, which is a pretty low bar (e.g., if none of the calls to `unlink()` specify`missing_ok`, then the assertion will pass so long as at least one call was made...).  Are you sure you don't want something like `any(ok)`?

Reply:
Yeah, the code got generalized when I added the parametrization, and I missed this check.